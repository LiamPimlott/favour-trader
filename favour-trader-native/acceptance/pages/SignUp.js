'use strict';

let I;

module.exports = {

    _init() {
        I = require('../../steps_file.js')();
    },

    register(firstname, lastname, email, password) {
        I.fillField('First Name', firstname);
        I.fillField('Last Name', lastname);
        I.fillField('E-mail Address', email);
        I.fillField('Password', password);
        I.tap('Sign Up');
        I.tap('OK');
    }
};
