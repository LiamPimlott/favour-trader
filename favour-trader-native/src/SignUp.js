import React, { Component } from 'react';
import { StyleSheet, View, Text, TextInput, TouchableOpacity, KeyboardAvoidingView, Image, ScrollView, Alert } from 'react-native';
import { StackNavigator } from 'react-navigation';

export default class SignUp extends React.Component {
	static navigationOptions = { 
		header: null 
	};
	render(){
		const { navigate } = this.props.navigation;
		
		return(
			<View style={styles.container}>
				
				<Image source={require('../images/test.jpg')} style={styles.backgroundImage}></Image>   
				<Text style={styles.title} >Ready to start trading?</Text>
				<Text style={styles.intro} >We just need a bit of info to get you going..</Text>  
				
				<KeyboardAvoidingView behavior="padding" style={styles.container}>
					<ScrollView style={ {top: '20%'} } contentContainerStyle={{justifyContent: 'center'}}> 
						<TextInput
							placeholder="First Name" 
							style={styles.input}
							ref={(input) => this.firstNameInput = input}
							onSubmitEditing={() => this.lastNameInput.focus()} 
							onChangeText={(firstName) => this.setState({firstName})}
						/>
						<TextInput
							placeholder="Last Name" 
							style={styles.inputSectionEnd}
							ref={(input) => this.lastNameInput = input}
							onChangeText={(lastName) => this.setState({lastName})}
							onSubmitEditing={() => this.emailInput.focus()} 
						/>
						<TextInput
							placeholder="e-mail address" 
							style={styles.input} 
							ref={(input) => this.emailInput = input}
							onChangeText={(email) => this.setState({email})}							
							onSubmitEditing={() => this.passwordInput.focus()} 
						/>
						<TextInput
							placeholder="Password"
							style={styles.input}
							secureTextEntry
							ref={(input) => this.passwordInput = input}
							onChangeText={(password) => this.setState({password})}
						/>
						<TouchableOpacity 
							style={styles.buttonContainer}
							onPress = { 
								() => {
									Alert.alert('Account Created', 'An account has been created under the email ' + this.state.email + '.  Happy Trading! :)'); 
									navigate('Login');
								}
							}  
						>
						
						<Text style={styles.buttonText}>SIGN UP</Text> 
						
						</TouchableOpacity>						
					</ScrollView>       
				</KeyboardAvoidingView>
			</View> 
		);
	}
}

const styles = StyleSheet.create({
	container: {  
		flex: 1,
		backgroundColor: 'transparent',
		alignItems: 'center'
	},    
	backgroundImage: {
		position: 'absolute',
		bottom: 0        
	},
	buttonContainer: { 
		backgroundColor: '#6F4B9B',
		justifyContent: 'center',
		height: '8%',
		marginTop: 10, 
		marginBottom: 5,
		borderRadius: 20
	},
	buttonText: {
		textAlign: 'center',
		fontSize: 25 
	},
	title: { 
		fontWeight: 'bold',
		fontSize: 40,
		top: '5%',
		marginBottom: 0
	},  
	intro: { 
		fontSize: 20,
		top: '5%',
		marginBottom: '5%'
	},
	sectionDetail: { 
		fontSize: 20,
		marginBottom: '3%'
	},
  	input: { 
		backgroundColor: '#9978C2',
		padding: 20,
		marginBottom: 5,      
		fontSize: 20,
		width: 400
	},
	inputSectionEnd: { 
		backgroundColor: '#9978C2',
		padding: 20,
		marginBottom: 20,      
		fontSize: 20,
		width: 400
	},
  
});
