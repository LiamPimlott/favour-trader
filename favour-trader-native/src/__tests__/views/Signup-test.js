import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import SignUp from '../../views/Signup.js';
import toJson from 'enzyme-to-json';
import renderer from "react-test-renderer"; //added this line

const Adapter = require("enzyme-adapter-react-16");
configure({adapter : new Adapter()});

let findById = function(tree, placeholder) {
    if(tree.props && tree.props.placeholder === placeholder) {
        return tree
    }
    if(tree.children && tree.children.length > 0)
    {
        let childs = tree.children
        for(let i = 0; i < childs.length; i++)
        {
            let item = findById(childs[i], placeholder)
            if(typeof(item) !== 'undefined') {
                return item
            }
        }
    }
}

describe('Testing SignUp component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <SignUp />
    );
   expect(toJson(wrapper)).toMatchSnapshot(); //edited this line
 });

    it('should render Valid input', () => {
        const navigation = { navigate: jest.fn() };
        let tree = renderer.create(<SignUp/>).toJSON()

        expect(findById(tree, 'E-mail Address')).toBeDefined()
        expect(findById(tree, 'Password')).toBeDefined()
        expect(findById(tree, 'First Name')).toBeDefined()
        expect(findById(tree, 'Last Name')).toBeDefined()
    })

    it('should render Empty input', () => {
        const navigation = { navigate: jest.fn() };
        let tree = renderer.create(<SignUp/>).toJSON()

        expect(findById(tree, '')).toBeUndefined()
    })

    it('should render Invalid input', () => {
        const navigation = { navigate: jest.fn() };
        let tree = renderer.create(<SignUp/>).toJSON()

        expect(findById(tree, 'Invalidate username ')).toBeUndefined()
        expect(findById(tree, 'Invalidate password')).toBeUndefined()
        expect(findById(tree, 'Invalidate last name')).toBeUndefined()
        expect(findById(tree, 'Invalidate first name')).toBeUndefined()
    })
});