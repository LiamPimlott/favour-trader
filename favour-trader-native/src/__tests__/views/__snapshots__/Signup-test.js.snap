// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SignUp component renders as expected 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  keyboardVerticalOffset={0}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Card
    title="Ready to do Favour â‡‹ Trading?"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      We just need a bit of info to get you going..
    </Text>
    <View
      style={
        Object {
          "marginBottom": 15,
          "marginTop": 10,
        }
      }
    >
      <Input
        leftIcon={
          <Icon
            color="black"
            name="face"
            raised={false}
            reverse={false}
            reverseColor="white"
            size={24}
            underlayColor="white"
          />
        }
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="First Name"
      />
    </View>
    <View
      style={
        Object {
          "marginBottom": 15,
          "marginTop": 10,
        }
      }
    >
      <Input
        leftIcon={
          <Icon
            color="black"
            name="face"
            raised={false}
            reverse={false}
            reverseColor="white"
            size={24}
            underlayColor="white"
          />
        }
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Last Name"
      />
    </View>
    <View
      style={
        Object {
          "marginBottom": 15,
          "marginTop": 10,
        }
      }
    >
      <Input
        leftIcon={
          <Icon
            color="black"
            name="mail-outline"
            raised={false}
            reverse={false}
            reverseColor="white"
            size={24}
            underlayColor="white"
          />
        }
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="E-mail Address"
      />
    </View>
    <View
      style={
        Object {
          "marginBottom": 15,
          "marginTop": 10,
        }
      }
    >
      <Input
        leftIcon={
          <Icon
            color="black"
            name="lock-outline"
            raised={false}
            reverse={false}
            reverseColor="white"
            size={24}
            underlayColor="white"
          />
        }
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 10,
          }
        }
      />
    </View>
    <Button
      TouchableComponent={[Function]}
      backgroundColor="#03A9F4"
      buttonStyle={
        Object {
          "borderRadius": 0,
          "marginBottom": 10,
          "marginLeft": 0,
          "marginRight": 0,
          "width": 300,
        }
      }
      clear={false}
      disabled={false}
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      title="Sign Up"
    />
  </Card>
</KeyboardAvoidingView>
`;
